.github/workflows/rust-workflow.yml

```diff
@@ -49,7 +49,8 @@ jobs:
     
     - name: Doc-test
       working-directory: head
-      run: cargo +${NIGHTLY_VERSION} test --doc --workspace --verbose
+      # Don't run doc-tests for binary crates
+      run: cargo metadata --format-version 1 --no-deps | jq '.packages[].targets[].crate_types[] | select(. == "lib")' -e && cargo +${NIGHTLY_VERSION} test --doc --workspace --verbose
```

Comment:
Is it possible to do `cargo +nightly test --doc --lib` to still test the library part of the binaries? (and to have an easier to understand command)

Reply:
Afraid not.
```shell
$ cargo +nightly test --doc --lib
error: Can't mix --doc with other target selecting options
```

Reply:
sad